hazelcast/src/main/java/com/hazelcast/config/NearCacheConfigAccessor.java

```diff
@@ -28,14 +28,19 @@
     private NearCacheConfigAccessor() {
     }
 
-    public static void initDefaultMaxSizeForOnHeapMaps(NearCacheConfig nearCacheConfig) {
+    public static NearCacheConfig copyWithInitializedDefaultMaxSizeForOnHeapMaps(NearCacheConfig nearCacheConfig) {
         if (nearCacheConfig == null) {
-            return;
+            return null;
         }
 
         EvictionConfig evictionConfig = nearCacheConfig.getEvictionConfig();
-        if (nearCacheConfig.getInMemoryFormat() != InMemoryFormat.NATIVE) {
-            evictionConfig.defaultSize = MapConfig.DEFAULT_MAX_SIZE;
+        if (nearCacheConfig.getInMemoryFormat() != InMemoryFormat.NATIVE && !evictionConfig.sizeConfigured) {
+            EvictionConfig copyEvictionConfig = new EvictionConfig(evictionConfig);
+            copyEvictionConfig.setSize(MapConfig.DEFAULT_MAX_SIZE);
+            NearCacheConfig copyNearCacheConfig = new NearCacheConfig(nearCacheConfig);
+            copyNearCacheConfig.setEvictionConfig(copyEvictionConfig);
+            return copyNearCacheConfig;
```

Comment:
minor: maybe a cosmetic change to increase readability:
```suggestion
        public static NearCacheConfig copyWithInitializedDefaultMaxSizeForOnHeapMaps(NearCacheConfig nearCacheConfig) {
        if (nearCacheConfig == null) {
            return null;
        }

        EvictionConfig evictionConfig = nearCacheConfig.getEvictionConfig();
        if (nearCacheConfig.getInMemoryFormat() == InMemoryFormat.NATIVE
                || evictionConfig.sizeConfigured) {
            return nearCacheConfig;
        }

        // create copy of eviction config
        EvictionConfig copyEvictionConfig = new EvictionConfig(evictionConfig)
                .setSize(MapConfig.DEFAULT_MAX_SIZE);

        // create copy of nearCache config and set eviction config
        return new NearCacheConfig(nearCacheConfig)
                .setEvictionConfig(copyEvictionConfig);
    }
```